editor/src/messages/tool/common_functionality/shape_editor.rs

```diff
@@ -50,6 +50,51 @@ pub struct ManipulatorPointInfo<'a> {
 
 pub type OpposingHandleLengths = HashMap<Vec<LayerId>, HashMap<ManipulatorGroupId, Option<f64>>>;
 
+const OUT_HANDLE: u8 = 0b001;
+const IN_HANDLE: u8 = 0b010;
+const BOTH_HANDLES: u8 = 0b011;
+const ANCHOR: u8 = 0b100;
+const ANCHOR_OUT: u8 = 0b101;
+const ANCHOR_IN: u8 = 0b110;
+const ALL: u8 = 0b111;
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct ManipulatorGroupData {
+	pub selection_type: u8,
+	pub group_id: ManipulatorGroupId,
+	pub layer_id: Vec<LayerId>,
+}
```

Comment:
Using Bitflags here will most likely yield worse performance here:
The struct consists of 64bit aligned fields and one 8bit aligned field -> 48bits of padding will be added anyway so we could just as well store 3 bool values on the struct directly and still use the same amount of meromy/cache lines (same memory throughput) but no longer need to spend cycles on the bitshift logic.
I think it would be beneficial to just use bools instead here to improve maintainability.
(and let me know if you have any questions about the performance aspects)